server/src/main/java/org/apache/iotdb/db/mpp/aggregation/FirstValueAccumulator.java

```diff
@@ -130,7 +130,6 @@ public void addStatistics(Statistics statistics) {
   @Override
   public void setFinal(Column finalResult) {
     reset();
-    hasCandidateResult = true;
     firstValue.setObject(finalResult.getObject(0));
```

Comment:
BTW, I think it's better to optimize this method in this pr to avoid unbox and box. You can refer to `addIntermediate` and `addInput`, using switch-case to get from exact type.
Remember to optimize all the implementation class of `Accumulator`.